okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -143,130 +299,339 @@ private static String checkOneDependency(ModuleDescriptor md, InterfaceDescripto
+              modsEnabled.remove(md.getId());
+              addTenantModule(tml, md.getId(), null, TenantModuleDescriptor.Action.disable);
+              return true;
+            }
+          }
+        }
+        String modules = entry.getValue().stream()
+            .map(x -> x.moduleDescriptor.getId())
+            .collect(Collectors.joining(", "));
+        errors.add(messages.getMessage("10213", modules, entry.getKey()));
       }
-      if (action == TenantModuleDescriptor.Action.enable && !modsEnabled.containsKey(id)) {
-        logger.info("getNextMT: return tm for action enable");
-        return tm;
+    }
+    return false;
+  }
+
+  private static boolean checkRequired(
+      Map<String, ModuleDescriptor> modsAvailable,
+      Map<String, ModuleDescriptor> modsEnabled, List<TenantModuleDescriptor> tml,
+      boolean fix, List<String> errors, Map<String, List<ModuleInterface>> providedInterfaces,
+      Map<String, List<ModuleInterface>> requiredInterfaces, Set<String> stickyModules) {
+
+    for (Map.Entry<String, List<ModuleInterface>> entry : requiredInterfaces.entrySet()) {
+      List<ModuleInterface> providedModuleInterfaces = providedInterfaces.get(entry.getKey());
+      if (providedModuleInterfaces == null) {
+        logger.info("Interface {} undefined and required", entry.getKey());
+        for (ModuleInterface req : entry.getValue()) {
+          if (fix) {
+            Map<String, ModuleDescriptor> modules =
+                checkInterfaceDepAvailable(modsAvailable, req.interfaceDescriptor);
+            if (modules.size() > 1) {
+              errors.add(messages.getMessage("10210", entry.getKey(), req.moduleDescriptor.getId(),
+                  String.join(", ", modules.keySet())));
+              return false;
+            } else if (!modules.isEmpty()) {
+              ModuleDescriptor mdFound = modules.values().iterator().next();
+              String id = req.moduleDescriptor.getId();
+              if (stickyModules.contains(req.moduleDescriptor.getId())) {
+                logger.info("Enable {}", mdFound.getId());
+                modsEnabled.put(mdFound.getId(), mdFound);
+                addTenantModule(tml, mdFound.getId(), null, TenantModuleDescriptor.Action.enable);
+                stickyModules.add(mdFound.getId());
+              } else {
+                logger.info("Removing {}", id);
+                modsEnabled.remove(id);
+                addTenantModule(tml, id, null, TenantModuleDescriptor.Action.disable);
+              }
+              return true;
+            }
+          }
+          errors.add(messages.getMessage("10211", entry.getKey(), req.moduleDescriptor.getId()));
+        }
       }
-      if (action == TenantModuleDescriptor.Action.disable && modsEnabled.containsKey(id)) {
-        logger.info("getNextTM: return tm for action disable");
-        return tm;
+    }
+    return false;
+  }
+
+  private static boolean checkCompatible(
```

Comment:
Can you add javadoc explaining the return value?

Reply:
done